LayoutTests/fast/reporting/reporting-observer-callback-does-not-leak.html

```diff
@@ -0,0 +1,49 @@
+<!DOCTYPE html>
+<html>
+<body>
+<iframe id="iframe"></iframe>
+<script src="../../resources/js-test.js"></script>
+<script>
+jsTestIsAsync = true;
+var frameDocumentID = 0;
+var checkCount = 0;
+
+description("Tests that ReportingObserver callbacks do not leak documents.");
+
+function iframeLoaded(frameDocument) {
+    frameDocumentID = internals.documentIdentifier(frameDocument);
+    iframe.addEventListener("load", () => {
+        handle = setInterval(() => {
+            gc();
+            if (!internals.isDocumentAlive(frameDocumentID)) {
+                clearInterval(handle);
+                testPassed("The iframe document didn't leak");
+                finishJSTest();
+            }
+            if (++checkCount > 500) {
+                clearInterval(handle);
+                testFailed("The iframe document leaked.");
+                finishJSTest();
+            }
+        }, 10);
```

Comment:
Since we have a stack scanning conservative GC, we should probably insert 10-20 iframes and observe that at least one of the iframes get collected. Otherwise, this test will become flaky.